src/GitHub.App/ViewModels/PullRequestDetailViewModel.cs

```diff
@@ -411,6 +415,30 @@ string GetBranchDisplayName(string targetBranchLabel)
             }
         }
 
+        string GetStatusDisplay(IPullRequestFileModel file, TreeChanges changes)
+        {
+            switch (file.Status)
+            {
+                case PullRequestFileStatus.Added:
+                    return "add";
+                case PullRequestFileStatus.Renamed:
+                    var fileName = file.FileName.Replace("/", "\\");
+                    var change = changes?.Renamed.FirstOrDefault(x => x.Path == fileName);
+
+                    if (change != null)
+                    {
+                        return Path.GetDirectoryName(change.OldPath) == Path.GetDirectoryName(change.Path) ?
+                            Path.GetFileName(change.OldPath) : change.OldPath;
+                    }
+                    else
+                    {
+                        return "rename";
+                    }
+                default:
+                    return null;
```

Comment:
The null is used in a trigger to hide the `TextBlock` that shows the status display: https://github.com/github/VisualStudio/blob/34a4071fe432e0daf0673c0772ca9b0137be9a5b/src/GitHub.VisualStudio/UI/Views/PullRequestDetailView.xaml#L360

I've run into problems before trying to make triggers work with empty strings, and indeed the suggested solution found here (and elsewhere) isn't pretty: http://stackoverflow.com/a/2582441/6448 . I can't remember the exact problem I've had in the past though - I can give it a try if you like?